nemo/collections/asr/parts/segment.py

```diff
@@ -123,21 +128,25 @@ def from_file(
         :param duration: duration in seconds when loading audio
         :return: numpy array of samples
         """
-        if isinstance(audio_file, (str, Path)):
-            ext = Path(audio_file).suffix
-        if not isinstance(audio_file, (str, Path)) or ext == ".wav":
-            with sf.SoundFile(audio_file, 'r') as f:
-                dtype = 'int32' if int_values else 'float32'
-                sample_rate = f.samplerate
-                if offset > 0:
-                    f.seek(int(offset * sample_rate))
-                if duration > 0:
-                    samples = f.read(int(duration * sample_rate), dtype=dtype)
-                else:
-                    samples = f.read(dtype=dtype)
-
-            samples = samples.transpose()
-        else:
+        samples = None
+        if not isinstance(audio_file, (str, Path)) or Path(audio_file).suffix.lower() in sf_supported_formats:
```

Comment:
Why not just use os.path.splitext() here instead of creating path objects. This line is called for every single sample in the training run, any efficiency issue is going to become a bottleneck.

Reply:
Ohhh, I don't know about splitext, I will change the code to use splitext